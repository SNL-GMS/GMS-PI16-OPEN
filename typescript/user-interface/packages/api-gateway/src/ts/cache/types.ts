import { UserActionDescription } from '@gms/common-model/lib/cache/types';
import Immutable from 'immutable';

/**
 * Specifies the type (method) for committing to the Global Cache.
 */
export type CommitFunc<T> = (items: T[]) => void;

/**
 * User Cache Processor interface.
 * Defines all of the common method definitions required
 * for a user cache item.
 */
// eslint-disable-next-line @typescript-eslint/naming-convention
export interface IUserCacheItem<T> {
  has(id: string): boolean;
  get(id: string): T;
  getAll(): T[];
  set(item: T): void;
  setAll(items: T[]): void;
  remove(item: T): void;
  removeAll(item: T[]): void;
  commitAll(): void;
  commitWithIds(ids: string[]): void;
  updateFromGlobalCache(cachedEntries: T[], overwrite: boolean): void;
}

/**
 * User Cache interface.
 * Defines all of the common method definitions required for a user cache.
 */
// eslint-disable-next-line @typescript-eslint/naming-convention
export interface IUserCache {
  getRedoPriorityOrder(id: string): number | undefined;
  getHistory(): History[];
  clearHistory(): void;
}

/**
 * User context for apollo and user sessions
 */
export interface UserContext {
  readonly sessionId: string;
  readonly userName: string;
  readonly userRole: string;
}

/**
 * Defines the interface of User Action.
 */
export interface UserAction {
  /**
   * The user action description
   */
  readonly description: UserActionDescription;

  /**
   * Returns the toString value of a user action.
   *
   * @returns the string representation of the action
   */
  toString(): string;
}

/**
 * Defines the hypothesis type for a hypothesis change.
 */
export enum HypothesisType {
  EventHypothesis = 'EventHypothesis',
  SignalDetectionHypothesis = 'SignalDetectionHypothesis'
}

/**
 * Represents hypothesis information for a change.
 * Defines a mapping from a hypothesis id to the main
 * object id.
 */
export interface HypothesisChangeInformation {
  readonly id: string;
  readonly hypothesisId: string;
  readonly type: HypothesisType;
  readonly parentId: string;
  readonly userAction: UserAction;
}

/**
 * Represents a history change.
 */
export interface HistoryChange {
  readonly id: string;
  readonly active: boolean;
  readonly eventId: string;
  readonly conflictCreated: boolean;
  readonly hypothesisChangeInformation: HypothesisChangeInformation;
}

/**
 * Represents the history.
 */
export interface History {
  readonly id: string;
  readonly description: UserActionDescription;
  readonly changes: Immutable.List<HistoryChange>;
}

/**
 * Invalid data; typically data that has been deleted.
 */
export interface InvalidData {
  readonly eventIds: string[];
  readonly signalDetectionIds: string[];
}
