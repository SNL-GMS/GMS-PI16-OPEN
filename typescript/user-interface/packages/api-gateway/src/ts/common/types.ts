import { CommonTypes } from '@gms/common-model';

/**
 * Common model definitions shared across gateway data APIs
 */

/**
 * Processing Context part of data structure that filtered waveform
 * control service uses to tell who/where the request came from. In our case
 * from Interactive UI user
 */
export interface ProcessingContext {
  readonly analystActionReference: AnalystActionReference;
  readonly processingStepReference: ProcessingStepReference;
  readonly storageVisibility: string;
}

/**
 * Analyst action reference
 */
export interface AnalystActionReference {
  readonly processingStageIntervalId: string;
  readonly processingActivityIntervalId: string;
  readonly analystId: string;
}

/**
 * Processing step reference
 */
export interface ProcessingStepReference {
  readonly processingStageIntervalId: string;
  readonly processingSequenceIntervalId: string;
  readonly processingStepId: string;
}

/**
 * Test data paths used when reading in data
 */
export interface TestDataPaths {
  readonly dataHome: string;
  readonly jsonHome: string;
  readonly additionalDataHome: string;
  readonly integrationDataHome: string;
}

/**
 * Represents calibration information associated with a waveform
 */
export interface ProcessingCalibration {
  readonly factor: number;
  readonly factorError: number;
  readonly period: number;
  readonly timeShift: number;
}

/**
 * Represents the configured type of data source the API Gateway provides access to - values:
 * Local - The API gateway loads data from local file storage for testing purposes
 * Service - The API gateway uses services to provide access to backend (e.g. OSD) data
 */
export enum AccessorDataSource {
  Local = 'Local',
  Service = 'Service'
}

/**
 * Enumerated list of source types used to compute distances to
 */
export enum DistanceSourceType {
  Event = 'Event',
  UserDefined = 'UserDefined'
}

/**
 * Represents input arguments for calculating distance measurement
 * relative to a specified source location
 */
export interface DistanceToSourceInput {
  // The type of the source the distance is measured to (e.g. and event)
  readonly sourceType: DistanceSourceType;

  // the unique ID of the source object
  readonly sourceId: string;
}

/**
 * Client log, object that describes the client log message
 */
export interface ClientLog {
  readonly logLevel: CommonTypes.LogLevel;
  readonly message: string;
  readonly time?: string;
}
