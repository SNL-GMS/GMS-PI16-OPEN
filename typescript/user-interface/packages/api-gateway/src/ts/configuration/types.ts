import { ConfigurationTypes } from '@gms/common-model';
import Immutable from 'immutable';

/**
 * Overall configuration
 */
export interface Configuration {
  /**
   * AnalystConfiguration keyed on user role
   */
  analystConfiguration: Immutable.Map<string, ConfigurationTypes.ProcessingAnalystConfiguration>;

  /**
   * SohConfigurationOld
   */
  sohConfiguration: ConfigurationTypes.SohConfiguration;

  /**
   * CommonConfiguration keyed on user role
   */
  commonConfiguration: Immutable.Map<string, ConfigurationTypes.ProcessingCommonConfiguration>;
}

/**
 * Default user role
 */
export const defaultUserRole = 'DEFAULT';
