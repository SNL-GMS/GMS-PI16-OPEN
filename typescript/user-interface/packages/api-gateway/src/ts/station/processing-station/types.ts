import { ChannelTypes, CommonTypes, ConfigurationTypes } from '@gms/common-model';

/**
 * Model definitions for the processing
 * StationGroup, Station, ChannelGroup and Channel data API
 */

/**
 * Encapsulates station-related data cached in memory
 */
export interface ProcessingStationData {
  stationGroupMap: Map<string, ProcessingStationGroup>;
  stationMap: Map<string, ProcessingStation>;
  channelGroupMap: Map<string, ProcessingChannelGroup>;
  channelMap: Map<string, ProcessingChannel>;
  sohStationGroupNameMap: Map<string, ConfigurationTypes.SOHStationGroupNameWithPriority[]>;
}
/**
 * Represents a group of stations used for monitoring.
 * This is the processing equivalent of the ReferenceNetwork.
 */
export interface ProcessingStationGroup {
  name: string;
  description: string;
  stations: ProcessingStation[];
}

/**
 * Represents an installation of monitoring sensors for the purposes of processing.
 * Multiple sensors can be installed at the same station.
 */
export interface ProcessingStation {
  name: string;
  type: CommonTypes.StationType;
  description: string;
  relativePositionsByChannel: Map<string, CommonTypes.Position>;
  location: CommonTypes.Location;
  channelGroups: ProcessingChannelGroup[];
  channels: ProcessingChannel[];
}

/**
 * ChannelGroupType enum represents the different groupings of channels
 */
export enum ChannelGroupType {
  PROCESSING_GROUP = 'PROCESSING_GROUP',
  SITE_GROUP = 'SITE_GROUP'
}

/**
 * Represents a physical installation (e.g., building, underground vault, borehole)
 * containing a collection of Instruments that produce Raw Channel waveform data.
 */
export interface ProcessingChannelGroup {
  name: string;
  description: string;
  location: CommonTypes.Location;
  type: ChannelGroupType;
  channels: ProcessingChannel[];
}

/**
 * Represents a source for unprocessed (raw) or processed (derived) time series data
 * from a seismic, hydroacoustic, or infrasonic sensor.
 */
export interface ProcessingChannel {
  name: string;
  canonicalName: string;
  description: string;
  station: string;
  channelDataType: ChannelTypes.ChannelDataType;
  channelBandType: ChannelTypes.ChannelBandType;
  channelInstrumentType: ChannelTypes.ChannelInstrumentType;
  channelOrientationType: ChannelTypes.ChannelOrientationType;
  channelOrientationCode: string;
  units: CommonTypes.Units;
  nominalSampleRateHz: number;
  location: CommonTypes.Location;
  orientationAngles: Orientation;
  configuredInputs: string[];
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  processingDefinition: Map<string, any>;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  processingMetadata: Map<ChannelProcessingMetadataType, any>;
}

/**
 * Represents the orientation angles used in processing channels
 */
export interface Orientation {
  horizontalAngleDeg: number;
  verticalAngleDeg: number;
}

/**
 * Represents the type of processing metadata values that can appear as keys in the
 */
export enum ChannelProcessingMetadataType {
  // General properties
  CHANNEL_GROUP = 'CHANNEL_GROUP',

  // Filtering properties
  FILTER_CAUSALITY = 'FILTER_CAUSALITY',
  FILTER_GROUP_DELAY = 'FILTER_GROUP_DELAY',
  FILTER_HIGH_FREQUENCY_HZ = 'FILTER_HIGH_FREQUENCY_HZ',
  FILTER_LOW_FREQUENCY_HZ = 'FILTER_LOW_FREQUENCY_HZ',
  FILTER_PASS_BAND_TYPE = 'FILTER_PASS_BAND_TYPE',
  FILTER_TYPE = 'FILTER_TYPE',

  // Channel steering properties (used in beaming, rotation)
  STEERING_AZIMUTH = 'STEERING_AZIMUTH',
  STEERING_SLOWNESS = 'STEERING_SLOWNESS',

  // Beaming properties
  BEAM_COHERENT = 'BEAM_COHERENT'
}
