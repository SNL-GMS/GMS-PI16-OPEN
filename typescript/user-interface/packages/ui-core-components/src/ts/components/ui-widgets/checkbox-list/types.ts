import { Position } from '@blueprintjs/core';
import { IconName } from '@blueprintjs/icons';
import Immutable from 'immutable';

export interface CheckboxListProps {
  checkboxEnum: any;
  enumToCheckedMap: Map<any, boolean>;
  enumKeysToDisplayStrings?: Map<string, string>;
  enumToColorMap?: Map<any, string>;
  onChange(map: Map<any, boolean>);
}

export interface SimpleCheckboxListProps {
  checkBoxListEntries: CheckboxListEntry[];
  onChange?(name: string): void;
}

export interface SimpleCheckboxListState {
  checkboxEntriesMap: Immutable.Map<string, CheckboxListEntry>;
}

export interface CheckboxListEntry {
  name: string;
  color?: string;
  iconButton?: IconButton;
  isChecked: boolean;
}

export interface IconButton {
  iconName: IconName;
  popover?: {
    position: Position;
    usePortal?: boolean;
    minimal?: boolean;
    content?: string | JSX.Element;
  };
  onClick?(event: React.MouseEvent<HTMLButtonElement>): void;
}
