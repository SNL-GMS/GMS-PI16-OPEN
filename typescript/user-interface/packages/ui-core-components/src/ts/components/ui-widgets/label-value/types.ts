import React from 'react';

/**
 * app determines the default styling for the label-value toolbar item, defaults to soh
 */
export interface LabelValueProps {
  value: string;
  label: string;
  tooltip: string;
  valueColor?: string;
  ianApp?: boolean;
  styleForValue?: React.CSSProperties;
  containerClass?: string;
}
