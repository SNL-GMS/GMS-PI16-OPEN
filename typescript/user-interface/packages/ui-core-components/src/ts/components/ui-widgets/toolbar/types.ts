import { IconName } from '@blueprintjs/core';
import { DateTimeFormat } from '@gms/common-util';
import React from 'react';

import { DateRangePickerTypes } from '../date-range-picker';

export interface ToolbarProps {
  items: ToolbarItem[];
  toolbarWidthPx: number;
  itemsLeft?: ToolbarItem[];
  minWhiteSpacePx?: number;
  spaceBetweenItemsPx?: number;
  hidden?: boolean;
  overflowIcon?: IconName;
}
export interface ToolbarState {
  indicesToOverflow: number[];
  leftIndicesToOverflow: number[];
  whiteSpaceAllotmentPx: number;
  checkSizeOnNextDidMountOrDidUpdate: boolean;
}

export type ToolbarItem =
  | NumericInputItem
  | DropdownItem
  | IntervalPickerItem
  | DateRangePickerItem
  | PopoverItem
  | SwitchItem
  | ButtonItem
  | ButtonGroupItem
  | LabelValueItem
  | CheckboxDropdownItem
  | LoadingSpinnerItem
  | CustomItem;

interface ToolbarItemBase {
  label?: string;
  tooltip: string;
  tooltipForIssue?: string;
  type: ToolbarItemType;
  rank: number;
  style?: React.CSSProperties;
  widthPx?: number;
  labelRight?: string;
  disabled?: boolean;
  icon?: IconName;
  onlyShowIcon?: boolean;
  menuLabel?: string;
  cyData?: string;
  hasIssue?: boolean;
  ianApp?: boolean; // If true, use default ian styling. defaults to false (SoH)
  onMouseEnter?(): void;
  onMouseOut?(): void;
}

interface NumericInputItem extends ToolbarItemBase {
  value: number;
  minMax: MinMax;
  step?: number;
  requireEnterForOnChange?: boolean;
  onChange(value: number);
}

interface DropdownItem extends ToolbarItemBase {
  dropdownOptions: any;
  dropdownText?: any;
  value: any;
  custom?: boolean;
  displayLabel?: boolean;
  onChange(value: any);
}

interface IntervalPickerItem extends ToolbarItemBase {
  startDate: Date;
  endDate: Date;
  defaultIntervalInHours: number;
  shortFormat?: boolean;
  onChange(startDate: Date, endDate: Date);
  onApplyButton(startDate: Date, endDate: Date);
}

interface DateRangePickerItem extends ToolbarItemBase {
  startTimeMs: number;
  endTimeMs: number;
  format: DateTimeFormat;
  durations?: DateRangePickerTypes.DurationOption[];
  minStartTimeMs?: number;
  maxEndTimeMs?: number;
  onChange(startTimeMs: number, endTimeMs: number);
  onApplyButton(startTimeMs: number, endTimeMs: number);
}

interface PopoverItem extends ToolbarItemBase {
  popoverContent: JSX.Element;
  onPopoverDismissed();
}

interface SwitchItem extends ToolbarItemBase {
  value: boolean;
  onChange(value: boolean);
}

interface ButtonItem extends ToolbarItemBase {
  onClick();
}

interface ButtonGroupItem extends ToolbarItemBase {
  buttons: ButtonItem[];
}

interface LabelValueItem extends ToolbarItemBase {
  value: string;
  valueColor?: string;
  styleForValue?: React.CSSProperties;
}

interface CheckboxDropdownItem extends ToolbarItemBase {
  values: Map<any, boolean>;
  colors?: Map<any, string>;
  enumKeysToDisplayStrings?: Map<string, string>;
  enumOfKeys: any;
  onChange(value: any): void;
  onPopUp?(ref?: HTMLDivElement): void;
  onPopoverDismissed?(): void;
}

interface LoadingSpinnerItem extends ToolbarItemBase {
  itemsToLoad: number;
  itemsLoaded?: number;
  hideTheWordLoading?: boolean;
  hideOutstandingCount?: boolean;
}

interface CustomItem extends ToolbarItemBase {
  element: JSX.Element;
}

export interface MinMax {
  min: number;
  max: number;
}

export enum ToolbarItemType {
  Switch = 'Switch',
  Popover = 'Popover',
  Dropdown = 'Dropdown',
  NumericInput = 'NumericInput',
  Button = 'Button',
  IntervalPicker = 'IntervalPicker',
  DateRangePicker = 'DateRangePicker',
  ButtonGroup = 'ButtonGroup',
  LabelValue = 'LabelValue',
  CheckboxList = 'CheckboxList',
  LoadingSpinner = 'LoadingSpinner',
  CustomItem = 'CustomItem'
}

export {
  ButtonGroupItem,
  ButtonItem,
  CheckboxDropdownItem,
  CustomItem,
  DateRangePickerItem,
  DateRangePickerTypes,
  DropdownItem,
  IntervalPickerItem,
  LabelValueItem,
  LoadingSpinnerItem,
  NumericInputItem,
  PopoverItem,
  SwitchItem
};
