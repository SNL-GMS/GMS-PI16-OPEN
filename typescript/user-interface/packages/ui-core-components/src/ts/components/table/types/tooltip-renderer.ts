/* eslint-disable @typescript-eslint/no-empty-interface */
import { ITooltipParams as AgITooltipParams } from 'ag-grid-community';

/** Wrapper interface class around ag-grid interface `ITooltipParams` */
export interface TooltipParams extends AgITooltipParams {}
