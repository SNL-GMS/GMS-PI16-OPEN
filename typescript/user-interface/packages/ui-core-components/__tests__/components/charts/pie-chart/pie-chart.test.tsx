/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable @typescript-eslint/no-magic-numbers */
import * as Enzyme from 'enzyme';
import * as React from 'react';
import { act } from 'react-dom/test-utils';

import { PieChart } from '../../../../src/ts/components/charts/pie-chart/pie-chart';
import * as PieChartTypes from '../../../../src/ts/components/charts/pie-chart/types';

const TIME_TO_WAIT_MS = 200;

/**
 * Fixes React warning that "An update to Component inside a test was not wrapped in act(...)."
 */
const waitForComponentToPaint = async (wrapper: any): Promise<void> => {
  // eslint-disable-next-line @typescript-eslint/await-thenable
  await act(async () => {
    await new Promise(resolve => setTimeout(resolve, TIME_TO_WAIT_MS));
    wrapper.update();
  });
};

const pieChartStyle: PieChartTypes.PieChartStyle = {
  diameterPx: 100,
  borderPx: 5
};

const props: PieChartTypes.PieChartProps = {
  style: pieChartStyle,
  percent: 40,
  className: 'string',
  pieSliceClass: 'string',
  status: 'string'
};

const wrapper = Enzyme.mount(<PieChart {...props} />);

describe('Pie Chart', () => {
  it('to be defined', () => {
    expect(PieChart).toBeDefined();
  });

  it('Pie Chart shallow renders', () => {
    const shallow = Enzyme.shallow(<PieChart {...props} />);
    expect(shallow).toMatchSnapshot();
  });

  it('Pie Chart renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('History List functions and clicks', async () => {
    const wrapper3 = Enzyme.mount(<PieChart {...props} />);

    wrapper3.setProps({ percent: 50 });
    await waitForComponentToPaint(wrapper3);
    expect(wrapper3.find('PieSlice')).toBeDefined();
  });
});
