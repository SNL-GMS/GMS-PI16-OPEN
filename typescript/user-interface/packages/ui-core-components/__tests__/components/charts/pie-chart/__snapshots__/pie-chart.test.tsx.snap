// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pie Chart Pie Chart renders 1`] = `
<Memo(CorePieChart)
  className="string"
  percent={40}
  pieSliceClass="string"
  status="string"
  style={
    Object {
      "borderPx": 5,
      "diameterPx": 100,
    }
  }
>
  <svg
    className="pie-chart string"
    height="110px"
    width="110px"
  >
    <g
      className="pie-chart__container"
    >
      <circle
        className="
            pie-chart
            string"
        cx={50}
        cy={50}
        r={55}
      />
      <PieSlice
        percent={40}
        pieSliceClass="string"
        style={
          Object {
            "borderPx": 5,
            "diameterPx": 100,
          }
        }
      >
        <circle
          className="pie-chart__pie-slice string"
          cx={50}
          cy={50}
          r={25}
          strokeDasharray="6283.185307179587 157.07963267948966 "
          strokeWidth={50}
        />
      </PieSlice>
    </g>
  </svg>
</Memo(CorePieChart)>
`;

exports[`Pie Chart Pie Chart shallow renders 1`] = `
<svg
  className="pie-chart string"
  height="110px"
  width="110px"
>
  <g
    className="pie-chart__container"
  >
    <circle
      className="
            pie-chart
            string"
      cx={50}
      cy={50}
      r={55}
    />
    <PieSlice
      percent={40}
      pieSliceClass="string"
      style={
        Object {
          "borderPx": 5,
          "diameterPx": 100,
        }
      }
    />
  </g>
</svg>
`;
